lib/index.js

```diff
@@ -3639,6 +3639,117 @@ _.wrapCallback = function (f) {
 };
 
 /**
+ * Takes an object or a constructor function and returns that object or
+ * constructor with streamified versions of its function properties.
+ * Passed constructors will also have their prototype functions
+ * streamified.  This is useful for wrapping many node style async
+ * functions at once, and for preserving those functions' context.
+ *
+ * @id streamifyAll
+ * @section Utils
+ * @name _.streamifyAll(source)
+ * @param {Object | Function} source - the function or object with
+ * node-style function properties.
+ * @api public
+ *
+ * var fs = _.streamifyAll(require('fs'));
+ *
+ * fs.readFileStream('example.txt').apply(function (data) {
+ *     // data is now the contents of example.txt
+ * });
+ */
+
+_.streamifyAll = function (arg) {
+    if (typeof arg !== 'function' && typeof arg !== 'object') {
+        throw new TypeError('takes an object or a constructor function');
+    }
+
+    // will not streamify inherited functions in ES3
+    var isES5 = (function () {
+      'use strict';
+      return Function.prototype.bind && !this;
+    }());
+    var suffix = 'Stream';
+
+    var getKeys = function (obj) {
+        var keys = [];
+        for (var k in obj) {
+            if (hasOwn.call(obj, k)) {
+                keys.push(k);
+            }
+        }
+        return keys;
+    };
+
+    var isClass = function (fn) {
+        if (!(typeof fn === 'function' && fn.prototype)) { return false; }
+        var allKeys = isES5 ? Object.getOwnPropertyNames : getKeys;
+        var keys = allKeys(fn.prototype);
+        return keys.length > 0 && !(keys.length === 1 &&
+                keys[0] === 'constructor');
+    };
+
+    var getInheritedKeys = function (obj){
+        var allProps = {};
+        var curr = obj;
+        var handleProp = function (prop) {
+            try {
+                if (typeof obj[prop] === 'function') {
+                    allProps[prop] = true;
+                }
+            } catch (e) {}
+        };
+        while (Object.getPrototypeOf(curr)) {
```

Comment:
As far I know I need to use Object.getPrototypeOf or __proto__ in order to support streamifying inherited methods, both of which are es5 features.  However, I do check for es5 support (which as you point out IE8 is missing) and I don't call this function if we're missing it.